components/eventing-controller/pkg/backend/nats/jetstream/jetstream_util.go

```diff
@@ -39,6 +44,16 @@ func toJetStreamRetentionPolicy(s string) (nats.RetentionPolicy, error) {
 	return nats.LimitsPolicy, fmt.Errorf("invalid stream retention policy %q", s)
 }
 
+func toJetStreamDiscardPolicy(s string) (nats.DiscardPolicy, error) {
+	switch s {
+	case DiscardPolicyOld:
+		return nats.DiscardOld, nil
+	case DiscardPolicyNew:
+		return nats.DiscardNew, nil
+	}
+	return nats.DiscardNew, fmt.Errorf("invalid stream discard policy %q", s)
```

Comment:
Did you mean to return `return nil, fmt.Errorf("invalid stream discard policy %q", s)`?

Reply:
at first it was as you wrote, but all the similar methods `toJetStreamRetentionPolicy`, `toJetStreamStorageType` return a default value (for unknown reasons)